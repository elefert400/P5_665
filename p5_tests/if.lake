bool b;
int main(){
    int x;
    int y;
    x = 1;
    y = 10;
    b = true;
    if(x == 1){}
    if(y != 10){}
    if(b){
        x = x + y;
    }
}