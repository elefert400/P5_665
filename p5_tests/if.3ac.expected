[BEGIN GLOBALS]
b
[END GLOBALS]
[BEGIN main LOCALS]
x (local)
y (local)
tmp0 (tmp)
tmp1 (tmp)
tmp2 (tmp)
[END main LOCALS]
enter main
x := 1
y := 10
b := 1
tmp0 := x EQ 1
iffalse tmp0 goto lbl_1
lbl_1: nop
tmp1 := y NEQ 10
iffalse tmp1 goto lbl_2
lbl_2: nop
iffalse b goto lbl_3
tmp2 := x ADD y
x := tmp2
lbl_3: nop
lbl_0: EXIT